### ISSUE TITLE
`AsDiscrete` needs to add channel dim for scalar label values

### ISSUE THREAD

**Issue by Nic-Ma** [Project-MONAI/MONAI#2554]
**Is your feature request related to a problem? Please describe.**
An question raised from the internal team:
Some PyTorch loss functions (like `torch.nn.CrossEntropyLoss`) require the label data has no channel dim, but postprocessing transforms and metrics require the label data with channel dim if multiple-classes. Need to add channel dim in post processing if ndim=0.

Actually, we had such feature in previous versions:
https://github.com/Project-MONAI/MONAI/blob/0.5.3/monai/transforms/post/array.py#L95

`VoteEnsemble` already considers the no-channel data:
https://github.com/Project-MONAI/MONAI/blob/0.6.0rc2/monai/transforms/post/array.py#L413

I think maybe we deleted it by mistake when developing the `decollate batch`.

**Comment by Nic-Ma** [comment]
I went through all the post transforms:
1. The input for `softmax` and `argmax` can't be single value, so it must have the channel dim.
2. The `OneHot` logic need input data to have channel dim.
3. Other transforms are for segmentation task which must contain multiple dims.

Thanks.
